coderd/database/dbauthz/dbauthz.go

```diff
@@ -2016,6 +2025,20 @@ func (q *querier) GetParameterSchemasByJobID(ctx context.Context, jobID uuid.UUI
 	return q.db.GetParameterSchemasByJobID(ctx, jobID)
 }
 
+func (q *querier) GetPrebuildMetrics(ctx context.Context) ([]database.GetPrebuildMetricsRow, error) {
+	if err := q.authorizeContext(ctx, policy.ActionRead, rbac.ResourceTemplate); err != nil {
```

Comment:
Self-review: make this rely on the prebuilds user/role. move initialization of collector into reconciler

Reply:
I'm working on the authz improvements, but lets not delay additional review until this is done. cc @dannykopping